app/src/androidTest/java/com/github/sdp_begreen/begreen/fragments/ContestCreationFragmentTest.kt

```diff
@@ -176,35 +258,109 @@ class ContestCreationFragmentTest {
     fun createContestStartDateIsDisplayed() {
         val date = "25/04/2030"
 
-        onView(withId(R.id.start_date_contest_text)).perform(replaceText(date)).check(matches(isDisplayed()))
+        onView(withId(R.id.start_date_contest_text)).perform(replaceText(date))
+            .check(matches(isDisplayed()))
         onView(withId(R.id.start_date_contest_text)).check(matches(withText(date)))
     }
 
     @Test
     fun createContestEndDateIsDisplayed() {
         val date = "25/04/2030"
 
-        onView(withId(R.id.end_date_contest_text)).perform(replaceText(date)).check(matches(isDisplayed()))
+        onView(withId(R.id.end_date_contest_text)).perform(replaceText(date))
+            .check(matches(isDisplayed()))
         onView(withId(R.id.end_date_contest_text)).check(matches(withText(date)))
     }
 
+    @Test
+    fun createContestDateIsChanging() {
+
+        val date = "25/04/2030"
+        val date2 = "26/04/2030"
+        val date3 = "27/04/2030"
+
+        runTest {
+            fragmentScenario.onFragment {
+                val vm by it.viewModels<ContestCreationViewModel>()
+            }
+        }
+
+        onView(withId(R.id.start_date_contest_text)).perform(click())
+        onView(withId(R.id.start_date_contest_text)).perform(replaceText(date))
+            .check(matches(isDisplayed()))
```

Comment:
Same as above, check display after perform replaceText

It comes two more time in the same test function as well.